src/libAtomVM/nifs.c

```diff
@@ -3419,6 +3426,37 @@ static term nif_ets_delete(Context *ctx, int argc, term argv[])
     }
 }
 
+static term nif_ets_update_counter(Context *ctx, int argc, term argv[])
+{
+    term ref = argv[0];
+    VALIDATE_VALUE(ref, is_ets_table_id);
+
+    term key = argv[1];
+    term operation = argv[2];
+    term default_value = term_invalid_term();
+    if (argc == 4) {
+        default_value = argv[3];
+        VALIDATE_VALUE(default_value, term_is_tuple);
+        term_put_tuple_element(default_value, 0, key);
+    }
```

Comment:
pure nitpicking here: let's write:
```c
term default_value;
if (argc == 4) {
[...]
} else {
    default_value = term_invalid_term();
}
```

So we underline that when 3 arguments are given, default_value is invalid.